### ISSUE TITLE
Collection first()-method order of callback arguments differ from other collection-methods (version 5.2.39)

### ISSUE THREAD

**Issue by SRautila** [laravel/framework#14548]
Short example:

`Psy Shell v0.7.2 (PHP 5.6.21 — cli) by Justin Hileman
>>> collect(["key" => "value"])->first(function($a, $b) { dump($a . " BEFORE " . $b); });
"key BEFORE value"
=> null
>>> collect(["key" => "value"])->map(function($a, $b) { dump($a . " BEFORE " . $b); });
"value BEFORE key"
=> Illuminate\Support\Collection {#756
     all: [
       "key" => null,
     ],
   }`

As you can see the order of the callback arguments are different for first compared to other collection-methods such as map and each. Is this intentional? The docs states that the order of arguments are first(function ($value, $key) which clearly is not the case.

**Comment by jhoff** [comment]
The [documentation](https://laravel.com/docs/5.2/collections#method-first) for `first` shows
```
->first(function ($key, $value) {
```

The [documentation](https://laravel.com/docs/5.2/collections#method-map) for `map` shows
```
->map(function ($item, $key) {
```

The [documentation](https://laravel.com/docs/5.2/collections#method-each) for `each` shows
```
->each(function ($item, $key) {
```

Those all match up with the code and is the correct behavior.

As to why the param order of `first` is different than the others, I'm not sure, but I would suspect that `map` and `each` used reversed params to remain inline with php's `array_map` and `array_walk` respectively.

**Comment by SRautila** [comment]
I noticed now that it's fixed in 5.3. Sorry for wasting your time.
